src/machines/commerce-machine.ts

```diff
@@ -81,9 +111,7 @@ export const commerceMachine = createMachine<
         initial: 'checkingCouponStatus',
         on: {
           SWITCH_PRICE: {
-            actions: assign({
-              priceId: (_context, event) => event.priceId,
-            }),
+            actions: ['switchPriceIdAndAssociatedData'],
```

Comment:
Adding a _switch price id_ action with a bit more logic so that we can check and adjust the applied coupon whenever the `priceId` changes.